docs/DeveloperGuide.md

```diff
@@ -2,37 +2,207 @@
+
+(Note: The lifelines of objects cannot be hidden after the X in the PlantUML software used to make this class diagram)
+
+
+### Ui Class
+The Ui class deals with taking in inputs from an input stream (in this case, the user input through `System.in`), 
+and then printing messages read-able by the user (through `System.out`). 
+
+### Recipe Class
+The Recipe class represents a singular recipe added by the user or loaded from the storage database. 
+A valid recipe object contains the following fields:
++ Name
++ List of Ingredients
++ List of Steps
++ Cuisine (optional)
++ Time taken (optional)
+
+The formatting of how a recipe is displayed and saved is handled by this class.
+
+### RecipeList Class
+The RecipeList class represents a list of any non-negative integer number of recipes. 
+This class handles the addition, deletion and editing of recipes.
+
+Shown below is a class diagram detailing the interaction between `RecipeList` and `Recipe`:
+![RecipeClassDiagram.png](img/ClassDiagrams/recipesClassDiagram.png)
+
+
+### Ingredient Class
+The ingredient class represents a singular ingredient added by the user or loaded from the storage database.
+A valid ingredient object contains the following field:
++ Name
+
+The formatting of how an ingredient is displayed and saved is handled by this class.
+
+### IngredientList Class
+The IngredientList class represents a list of any non-negative integer number of ingredients.
+This class handles the addition, deletion and editing of ingredients.
+
+Shown below is a class diagram detailing the interaction between `IngredientList` and `Ingredient`:
+![ingredientClasses.png](img/ClassDiagrams/ingredientsClassDiagram.png)
+
+
+### Command Classes
+The command class deals with all the possible commands accepted by YMFC, with each command representing a specific 
+recognised user input (E.g. ListCommand representing the command to list all the recipes in the recipeList).
 
+#### 1. The `Command` base class
+- Serves as the abstract parent class for all specific commands
+- Defines the basic structure and contract that each command class must follow
+- Each command performs a unique operation and can be expanded to include more functionality as required by YMFC
+- Standardises command behavior and establishes an inheritance hierarchy, allowing for the seamless addition and usage 
+of command functionalities
+- Core Elements:
+  - `isBye`: A flag that tracks if an exit command has been issued, following which programme should be terminated
+    - Set by calling `setBye() `
+    - Checked with `isBye()`
+  - `execute()`: An abstract method that must be implemented by all subclasses
+    - Defines the primary action performed by the command (E.g. add new recipe to `recipeList` for `AddRecipeCommand`)
+
+Shown below is a class diagram detailing the interaction between the parent `Command` class and it's child classes:
+![commandClasses.png](img/ClassDiagrams/commandsClassDiagram.png)
```

Comment:
Would it be possible to break this class diagram down or format it so that it is more easily viewed?